woocommerce/woocommerce-gateway-stripe
# Change payment method order in Stripe settings page

Fixes #2735 

## Changes proposed in this Pull Request:
This PR implements the reorder feature for the payment methods when UPE is disabled. By clicking on the "Change display order" button, the payment method list becomes draggable and you can drag and drop the payment methods to reorder them.

Some main changes of this PR are-
- Used `Reorder` from `framer-motion` to implement the sortable list.
- Introduces a new option `stripe_legacy_method_order` in Stripe settings object to hold the ordered list of legacy payment methods. 

## Testing instructions
- Run `npm install` to install `framer-motion`.
- Disable UPE and go to `Payment Methods` tab. At this point, the payment method list are not draggable.
- Click the `Change display order`. The checkboxes and `Customize` buttons will disappear and the list will become draggable.
- Drag and drop any method from the list and click `Save display order` button. You should find a POST request to save the new order in the Network tab.
- Refresh the page, the previous change should be sustained.
- Now change the order again but click `Cancel` instead of saving. The list should appear in the previous order.
- Enable UPE. With UPE enabled, the change order button does not appear and the list is not sortable.

### Note
- The [dismissible banner](https://href.li/?https://www.figma.com/file/xVAk98kbRKUByVTrhPc98g/?node-id=603-28309) is not implemented in this PR.
